ambry-replication/src/main/java/com/github/ambry/replication/ReplicationManager.java

```diff
@@ -210,20 +229,27 @@ public boolean removeReplica(ReplicaId replicaId) {
     List<RemoteReplicaInfo> remoteReplicaInfos = new ArrayList<>();
     PartitionId partition = replicaId.getPartitionId();
     Store store = storeManager.getStore(partition);
-    for (ReplicaId remoteReplica : peerReplicas) {
-      // We need to ensure that a replica token gets persisted only after the corresponding data in the
-      // store gets flushed to disk. We use the store flush interval multiplied by a constant factor
-      // to determine the token flush interval
-      FindToken findToken =
-          this.tokenHelper.getFindTokenFactoryFromReplicaType(remoteReplica.getReplicaType()).getNewFindToken();
-      RemoteReplicaInfo remoteReplicaInfo = new RemoteReplicaInfo(remoteReplica, replicaId, store, findToken,
-          TimeUnit.SECONDS.toMillis(storeConfig.storeDataFlushIntervalSeconds) * Replication_Delay_Multiplier,
-          SystemTime.getInstance(), remoteReplica.getDataNodeId().getPortToConnectTo());
-      replicationMetrics.addMetricsForRemoteReplicaInfo(remoteReplicaInfo, trackPerPartitionLagInMetric);
-      remoteReplicaInfos.add(remoteReplicaInfo);
-    }
-    if (replicationConfig.replicationTrackPerPartitionLagFromRemote) {
-      replicationMetrics.addLagMetricForPartition(partition);
+    try {
+      for (ReplicaId remoteReplica : peerReplicas) {
+        // We need to ensure that a replica token gets persisted only after the corresponding data in the
+        // store gets flushed to disk. We use the store flush interval multiplied by a constant factor
+        // to determine the token flush interval
+        FindToken findToken =
+            this.tokenHelper.getFindTokenFactoryFromReplicaType(remoteReplica.getReplicaType()).getNewFindToken();
+        RemoteReplicaInfo remoteReplicaInfo = null;
+        remoteReplicaInfo = new RemoteReplicaInfo(remoteReplica, replicaId, store, findToken,
+            TimeUnit.SECONDS.toMillis(storeConfig.storeDataFlushIntervalSeconds) * Replication_Delay_Multiplier,
+            SystemTime.getInstance(), remoteReplica.getDataNodeId().getPortToConnectTo(),
+            storeKeyConverterFactory.getStoreKeyConverter());
+
+        replicationMetrics.addMetricsForRemoteReplicaInfo(remoteReplicaInfo, trackPerPartitionLagInMetric);
+        remoteReplicaInfos.add(remoteReplicaInfo);
+      }
+      if (replicationConfig.replicationTrackPerPartitionLagFromRemote) {
+        replicationMetrics.addLagMetricForPartition(partition);
+      }
+    } catch (Exception e) {
+      logger.error("Encountered exception instantiating RemoteReplicaInfos", e);
```

Comment:
Sure, you are right, may be it is not ok to end up with partial list of remoteReplicaInfos. Removed try/catch block for handling the exception now. Previously, added this to handle exception from instantiation of StoreKeyConverter object being passed (added in this PR) to RemoteReplicaInfo constructor. Made changes to avoid passing StoreKeyConverter to RemoteReplicaInfo and instead directly store the converted keys in RemoteReplicaInfo.exchageMetadataResponse.remoteKeyToLocalKeyMap.